net/ip.h

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: 2023 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef NET_IP_H_
+#define NET_IP_H_
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace net {
+
+struct IP4Address {
```

Comment:
Do these need to be classes? As they are right now, they feel like a roundabout way of doing something like `std::string as_ipv4_string(std::uint32_t);`?